luigi/worker.py

```diff
@@ -248,9 +249,23 @@ def _recursive_terminate(self):
                 try:
                     child.terminate()
                 except psutil.NoSuchProcess:
+                    # Process is already gone, no one else has claimed this PID yet
```

Comment:
can you also add `if super(TaskProcess, self).is_alive():` wrapping the whole function?  like right above the import psutil.  this will check if the original processes spawned off is even still alive.